### PR TITLE
[query/service] add docs, use consistent branding

### PR DESCRIPTION
**Pull request by danking**
Writing the ATGU presentation made clear that we need consistent branding. I am
going with "Hail Query-on-Batch" and the "Batch backend".

### CODE REVIEW

File: hail/python/hail/context.py [hunk 1]
```diff
@@ -219,20 +224,22 @@ def init(sc=None, app_name='Hail', master=None, local='local[*]',
     Parameters
     ----------
     sc : pyspark.SparkContext, optional
-        Spark context. By default, a Spark context will be created.
+        Spark Backend only. Spark context. If none is specified, the Spark backend
+        will create a Spark context will be created.
```
Comment (jigold) [inline hail/python/hail/context.py:228]: wording here.

File: hail/python/hail/docs/cloud/query_on_batch.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+===================
+
+.. warning::
+
+    Hail Query-on-Batch (the Batch backend) is currently in beta. This means some functionality is
+    not yet working. Please `contact us <https://discuss.hail.is>`__ if you would like to use missing
+    functionality on Query-on-Batch!
+
+
+Hail Query-on-Batch uses Hail Batch instead of Apache Spark. Instead of a Dataproc cluster, you will
```
Comment (jigold) [inline hail/python/hail/docs/cloud/query_on_batch.rst:12]: `Hail Query-on-Batch uses Hail Batch instead of Apache Spark to execute jobs.`

**Review by jigold — DISMISSED** [review]

**Comment by jigold** [comment]
Does the query Makefile need to change at all?

**Comment by danking** [comment]
good point yes

**Review by jigold — APPROVED** [review]

**Pull request merged**